src/core/geometry/qgspointv2.cpp

```diff
@@ -462,3 +462,23 @@ double QgsPointV2::azimuth( const QgsPointV2& other ) const
   double dy = other.y() - mY;
   return ( atan2( dx, dy ) * 180.0 / M_PI );
 }
+
+
+QgsPointV2 QgsPointV2::project( double distance, double azimuth, double inclinaison ) const
+{
+    double rads_xy = azimuth * M_PI / 180.0;
+    if ( !is3D() )
```

Comment:
I also thought of several ways to implement this method.

- distinguish a function for a 2D point (as a project2D method) and one for a 3D point (as a project3D method)
- consider that if we pass a 2D point then there is a projection only in 2D (the case for QgsPoint.project) and if we have a 3D point then we project in 3D (my proposal)
- consider that if we pass a point has necessarily a projection including in 3D (your expectation)

I did not propose the latter because it will force always to return a PointZ which for my few cases is not necessarily requested or desired. So you have to addZ for your QgsPointV2 (2D) before use project for have 3D projection.